### ISSUE TITLE
Regression: grid training strategy makes training sets overlap

### ISSUE THREAD

**Issue by de-souza** [SciML/NeuralPDE.jl#611]
Before merging #578 (commit 15fece6), the grid training strategy was generating PDE training sets that did share coordinates with the boundary training sets. Since this PR was merged, this is not the case anymore.

#### Old Behavior

1. Install NeuralPDE one commit before 15fece6:

```julia
using Pkg
Pkg.add(; name="NeuralPDE", rev="15fece6~")
```

2. Generate training sets:

```julia
using NeuralPDE

@parameters x
@variables u(..)

eqs = [u(x) ~ 0]
bcs = [u(0) ~ 0]
domains = [x ∈ (0, 1)]
dx = [0.2]

train_sets = generate_training_sets(domains, dx, eqs, bcs, Float64, [x], [u])
pde_set = train_sets[1][1]
bcs_set = train_sets[2][1]

@show pde_set bcs_set
```

3. Output:

```julia
pde_set = [0.2 0.4 0.6 0.8 1.0]
bcs_set = [0.0;;]
```

The coordinate x = 0.0 is only included in the boundary training set.

#### New Behavior

1. Install NeuralPDE at commit 15fece6:

```julia
using Pkg
Pkg.add(; name="NeuralPDE", rev="15fece6")
```

2. Restart Julia and generate training sets.

3. Output:

```julia
pde_set = [0.0 0.2 0.4 0.6 0.8 1.0]
bcs_set = [0.0;;]
```

The coordinate x = 0.0 is included in both the boundary training set the PDE training set.

**Comment by ChrisRackauckas** [comment]
Interesting. Is there a case where this makes a substantive difference? The only technically needs to be solved in the interior, indeed, but I'm curious if there's a case where this actually makes a difference for which the BC still gives a well-defined solution. Since if the BC requires the PDE to not be satisfied "infinitesimally" close to the edge, then the solution wouldn't be well-defined, in which case sampling and adding it to the loss shouldn't necessarily be an issue?
